File: frontend/src/types/ticket.ts
export const TicketStatus = {
    Todo: 0,
    InProgress: 1,
    Done: 2
} as const;

export type TicketStatus =
    typeof TicketStatus[keyof typeof TicketStatus];

export const TicketPriority = {
    Low: 0,
    Medium: 1,
    High: 2
} as const;

export type TicketPriority =
    typeof TicketPriority[keyof typeof TicketPriority];

export interface Ticket {
    id: number;
    title: string;
    description: string;
    status: TicketStatus;
    priority: TicketPriority;
    assignedUser?: string;
}

export interface CreateTicketRequest {
    title: string;
    description: string;
    scrumboardId: number;
    priority: TicketPriority;
}

export interface UpdateTicketRequest {
    title: string;
    description: string;
    priority: TicketPriority;
}

File: backend/WorkloadPlanner/Models/Ticket.cs
using System.ComponentModel.DataAnnotations.Schema;
using WorkloadPlanner.Enums;

namespace WorkloadPlanner.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        [Column(TypeName = "nvarchar(20)")]
        public TicketStatus Status { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public TicketPriority Priority { get; set; }
        public DateTime CreatedAt { get; set; }
        public int ScrumBoardId { get; set; }
        public ScrumBoard ScrumBoard { get; set; } = null!;
        public string? AssignedUserId { get; set; }
        public ApplicationUser? AssignedUser { get; set; }
        public string CreatedById { get; set; } = string.Empty;
        public ApplicationUser CreatedBy { get; set; } = null!;
    }
}